test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,221 @@
+        },
+        "hugetlb" : {
+            "enabled"         : false
+        },
+        "memory":
+        {
+            "enabled"         : true,
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : [],
+            "soft_limit"      : false,
+            "default"         : "256MB",
+            "reserve_percent" : "0",
+            "reserve_amount"  : "0MB"
+        },
+        "memsw":
+        {
+            "enabled"         : true,
+            "exclude_hosts"   : [],
+            "exclude_vntypes" : [],
+            "default"         : "256MB",
+            "reserve_percent" : "0",
+            "reserve_amount"  : "128MB"
+        }
+    }
+}"""
+
+        self.paths = self.get_paths()
+        if not self.paths:
+            self.skipTest("No cgroups mounted")
+        self.server.set_op_mode(PTL_CLI)
+        self.server.cleanup_jobs(extend='force')
+        self.remove_vntype()
+        Job.dflt_attributes[ATTR_k] = 'oe'
+        # Configure the scheduler to schedule using vmem
+        a = {'resources': 'ncpus,mem,vmem,host,vnode'}
+        self.scheduler.set_sched_config(a)
+        # Import the hook
+        self.hook_name = 'pbs_cgroups'
+        self.hook_file = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                      'lib',
+                                      'python',
+                                      'altair',
+                                      'pbs_hooks',
+                                      'pbs_cgroups.PY')
+        self.load_hook(self.hook_file)
+        # Enable the cgroups hook
+        conf = {'enabled': 'True', 'freq': 2}
+        self.server.manager(MGR_CMD_SET, HOOK, conf, self.hook_name)
+        # Restart mom so exechost_startup hook is run
+        self.mom.signal('-HUP')
+
+    def remove_vntype(self):
+        """
+        Unset the vnode type for the local mom.
+        """
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
+        self.logger.info("Deleting vntype files from moms")
+        r = self.du.rm(path=vntype_file, force=True, sudo=True, logerr=False)
+        if not r:
+            self.skipTest("pbs_cgroups_hook: need root privileges")
+
+    @staticmethod
+    def get_paths(self):
+        """
+        Returns a dictionary containing the location where each cgroup
+        is mounted.
+        """
+        paths = {}
+        # Loop through the mounts and collect the ones for cgroups
+        with open(os.path.join(os.sep, "proc", "mounts"), 'r') as fd:
+            for line in fd:
+                entries = line.split()
+                if entries[2] != "cgroup":
+                    continue
+                flags = entries[3].split(',')
+                subsys = os.path.basename(entries[1])
+                paths[subsys] = entries[1]
+                if 'memory' in flags:
+                    paths['memsw'] = paths['memory']
```

Comment:
While running the original tests for Cray I noticed memory subsystem is called mcgroup. Hence I suggest changing this to following
paths['memsw] = paths[subsys]
and add following as well
paths['memory'] = paths[subsys]

Reply:
Done